pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -395,20 +396,14 @@ func (r *ForeignClusterReconciler) checkIncomingPeeringStatus(ctx context.Contex
 	remoteClusterID := foreignCluster.Spec.ClusterIdentity.ClusterID
 	localNamespace := foreignCluster.Status.TenantNamespace.Local
 
-	var incomingResourceRequestList discoveryv1alpha1.ResourceRequestList
-	if err := r.Client.List(ctx, &incomingResourceRequestList, client.HasLabels{
-		liqoconst.ReplicationStatusLabel}, client.MatchingLabels{
-		liqoconst.ReplicationOriginLabel: remoteClusterID,
-	}); err != nil {
-		klog.Error(err)
-		return err
+	incomingResourceRequestList, err := resourcerequestoperator.GetResourceRequests(ctx, r.Client, remoteClusterID)
+	if err != nil {
+		return fmt.Errorf("reading resource requests for %v: %w", remoteClusterID, err)
 	}
 
-	status, reason, message, err := getPeeringPhaseList(foreignCluster, &incomingResourceRequestList)
+	status, reason, message, err := getPeeringPhaseList(foreignCluster, incomingResourceRequestList)
 	if err != nil {
-		err = fmt.Errorf("[%v] %w in namespace %v", remoteClusterID, err, localNamespace)
-		klog.Error(err)
-		return err
+		return fmt.Errorf("[%v] %w in namespace %v", remoteClusterID, err, localNamespace)
```

Comment:
we should be consistent in logs

I think that is nice to have each log starting with `[remoteClusterID] ...` (as in the last error, line 406)

Reply:
Fair point, but I think that should be handled in logs specifically while errors are more general and should follow the usual conventions - I'll add the `[remoteClusterID]` prefix in `klog.Error` in the caller.